### PR TITLE
docs: Add Slack example

### PR DESCRIPTION
**Pull request by vdmgolub**

### CODE REVIEW

File: examples/slack/README.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+3. Add the following yaml spec to `repos.yaml` or `atlantis.yaml` config files:
+   ```yaml
+   repos:
+     - id: /.*/
+       workflow: terraform-infracost
+       post_workflow_hooks:
+         # Clean up any files left over from the run
+         - run: |
+             rm /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-infracost.json
+             rm /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-slack-message.json
```
Comment (vdmgolub) [inline examples/slack/README.md:31]: @hugorut Is this step working well for you with `$WORKSPACE-$REPO_REL_DIR` in the file name? It keeps failing for me as these values are not available in the post workflow hook. I removed it in this example.

File: examples/slack/README.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+This `repo.yaml` file contains a workflow specification to send Infracost cost estimates to Slack.
+
+Slack message blocks have a 3000 char limit so the Infracost CLI automatically truncates the middle of `slack-message` output format.
+
+## Table of Contents
+
+* [Running with GitHub](#running-with-github)
+* [Running with GitLab](#running-with-gitlab)
+* [Running with Azure Repos](#running-with-azure-repos)
```
Comment (vdmgolub) [inline examples/slack/README.md:11]: Added ToC and the title on line 1. If it's 👍 I'll go over other examples and do the same.

File: examples/slack/README.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+         steps:
+           - init
+           - plan
+           - show # this writes the plan JSON to $SHOWFILE
+           # Run Infracost breakdown and save to a tempfile, namespaced by this project, PR, workspace and dir
+           - run: |
+               infracost breakdown --path=$SHOWFILE \
+                                   --format=json \
+                                   --log-level=info \
+                                   --out-file=/tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-infracost.json
```
Comment (vdmgolub) [inline examples/slack/README.md:44]: Should we use this multiline formatting everywhere? If 👍 I'll go over and update other examples.

File: examples/slack/README.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+           # Use Infracost comment to create a comment containing the results for this project
+           - run: |
+               infracost comment github --repo $BASE_REPO_OWNER/$BASE_REPO_NAME \
+                                        --pull-request $PULL_NUM \
+                                        --path /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-infracost.json \
+                                        --behavior new \
```
Comment (vdmgolub) [inline examples/slack/README.md:50]: As agreed, the default behavior is `new`. I'll update it in other examples when add cases for Azure Repos.

File: examples/slack/README.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+   ```
+3. Add the following yaml spec to `repos.yaml` or `atlantis.yaml` config files:
+   ```yaml
+   repos:
+     - id: /.*/
+       workflow: terraform-infracost
+       post_workflow_hooks:
+         # Clean up any files left over from the run
+         - run: |
+             rm /tmp/${BASE_REPO_OWNER//\//-}-$BASE_REPO_NAME-$PULL_NUM-infracost.json
```
Comment (vdmgolub) [inline examples/slack/README.md:153]: Have to do this `${BASE_REPO_OWNER//\//-}` for Azure Repos - the owner comes in as `infracost/base` (using our demo repo) and it fails to create files.

File: examples/slack/README.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+                 echo "Not posting to Slack since cost change is zero"
+                 exit 0
+               fi
+
+               if [ -z "$SLACK_WEBHOOK_URL" ]; then
+                 echo "No \$SLACK_WEBHOOK_URL variable set."
+                 exit 1
+               fi
+
+               curl -X POST -H "Content-type: application/json" -d @/tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-slack-message.json $SLACK_WEBHOOK_URL
```
Comment (vdmgolub) [inline examples/slack/README.md:71]: Have to use cURL, as the native Atlantis Slack hook is available only for `apply` event: https://www.runatlantis.io/docs/using-slack-hooks.html#configuring-slack-for-atlantis

File: examples/slack/README.md (hunk 1 above)
Comment (hugorut) [reply examples/slack/README.md:31]: Nope, once I changed to `post_workflow_hook` this started failing - I'm bringing a PR to remove all of these from the examples as it's not needed

**Review by hugorut — APPROVED** [review]
LGTM

**Review by aliscott — APPROVED** [review]
LGTM

**Pull request merged**